assets/js/base/components/text-input/style.scss

```diff
@@ -64,9 +64,15 @@
 	}
 
 	&.has-error input {
-		border-color: $error-red;
+		&,
+		&:hover,
+		&:focus,
+		&:active {
+			border-color: $error-red;
+		}
 		&:focus {
-			outline-color: $error-red;
+			outline: 1px dotted $error-red;
+			outline-offset: 2px;
```

Comment:
_Maybe_ (soft maybe!) these could be bundled up in a mixin? Probably no need until they spread a bit further.